spaghetti/util.py

```diff
@@ -718,9 +719,10 @@ def _points_as_gdf(
             pts_dict = pp.snapped_coordinates
 
     # instantiate geopandas.GeoDataFrame
-    pts_list = list(pts_dict.items())
-    points = geopandas.GeoDataFrame(pts_list, columns=[id_col, geom_col])
-    points.geometry = points[geom_col].map(lambda p: Point(p))
+    pts = numpy.asarray(list(pts_dict.values()))
+    points = geopandas.GeoDataFrame(
+        pts_dict.keys(), columns=[id_col], geometry=shapely.points(pts[:, 0], pts[:, 1])
```

Comment:
```suggestion
        pts_dict.keys(), columns=[id_col], geometry=shapely.points(pts)
```
This should be enough. Otherwise looking good. It requires shapely 2.0 but now that both libpysal and esda will also want it, it should not be an issue here.

Reply:
OK, I'll go ahead and squeeze that into a one-liner since it's deep in the code base. Also, I'll need to update the reqs, etc.